coalib/testing/LocalBearTestHelper.py

```diff
@@ -26,20 +26,45 @@ def execute_bear(bear, *args, **kwargs):
     return list(bear_output_generator)
 
 
+def get_local_bear_run_settings_dict(local_bear, settings_dict):
+    local_bear_metadata = local_bear.get_metadata()
+    compat_settings_dict = {key: settings_dict[key] for key in settings_dict
+                            if key in local_bear_metadata.optional_params}
+    return compat_settings_dict
+
+
 def get_results(local_bear,
                 lines,
                 filename=None,
                 force_linebreaks=True,
                 create_tempfile=True,
                 tempfile_kwargs={},
                 settings={}):
+    deps_results = dict()
+    if local_bear.BEAR_DEPS:
+        # Get results of bear's dependencies first
+        for bear in local_bear.BEAR_DEPS:
+            uut = bear(local_bear.section, queue.Queue())
+            deps_results[bear.name] = get_results(uut,
+                                                  lines,
+                                                  filename,
+                                                  force_linebreaks,
+                                                  create_tempfile,
+                                                  tempfile_kwargs,
+                                                  settings)
+
     with prepare_file(lines, filename,
                       force_linebreaks=force_linebreaks,
                       create_tempfile=create_tempfile,
-                      tempfile_kwargs=tempfile_kwargs) as (file, fname), \
-        execute_bear(local_bear, fname, file,
```

Comment:
You should be able to do:

```diff
-execute_bear(local_bear, fname, file,
+execute_bear(local_bear, fname, file, dependency_results=dep_results,
```

And initialise `if not local_bear.BEAR_DEPS: dep_results = None`.

If that doesnt work, IMO `Bear` needs to be enhanced to detect kwarg `dependency_results` and re-route that argument so it works correctly.